framework/web/Request.php

```diff
@@ -162,7 +162,39 @@ class Request extends \yii\base\Request
      * @see getBodyParams()
      */
     public $parsers = [];
+    /**
+     * @var array The configuration for trusting security related header.
+     * The array key is a regular expression for matching the hostname.
+     * The value is a list of headers to trust.
+     * Example 1: trust all headers from domains ending in '.trusted.com'
+     * Example 2: trust just the x-forwarded-for header from domains ending in '.partial.com'
+     * ```
+     * [
+     *     '/^.*\.trusted.com$/',
+     *     '/^.*\.partial.com$/' => ['X-Forwarded-For']
```

Comment:
Most proxies will ignore headers that they do not know about. So a proxy that doesn't set `X-Forwarded-Proto` but does set `X-Forwarded-For` will pass the other one through if the client sends it. This is insecure.